src/pages/index.js

```diff
@@ -0,0 +1,300 @@
+  }
+
+  // Like/Dislike logic
+  cardLikeButton.addEventListener("click", () => {
+    const isLiked = cardLikeButton.classList.contains(
+      "card__like-button_liked"
+    );
+
+    if (isLiked) {
+      api
+        .dislikeCard(data._id)
+        .then(() => {
+          cardLikeButton.classList.remove("card__like-button_liked");
+        })
+        .catch((err) => {
+          console.error("Error disliking card:", err);
+        });
+    } else {
+      api
+        .likeCard(data._id)
+        .then(() => {
+          cardLikeButton.classList.add("card__like-button_liked");
+        })
+        .catch((err) => {
+          console.error("Error liking card:", err);
+        });
+    }
+  });
+
+  // Delete card logic
+  cardDeleteButton.addEventListener("click", () => {
+    handleDeleteCardSubmit(cardElement, data._id);
+  });
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageElement.src = data.link;
+    previewModalCaptionElement.textContent = data.name;
+    previewModalImageElement.alt = data.name;
+  });
+
+  return cardElement;
+}
+
+function closeOverlay(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function handleEsc(evt) {
+  if (evt.key === "Escape") {
+    const openModal = document.querySelector(".modal_opened");
+    if (openModal) {
+      closeModal(openModal);
+    }
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  modal.addEventListener("click", closeOverlay);
+  document.addEventListener("keydown", handleEsc);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  modal.removeEventListener("click", closeOverlay);
+  document.removeEventListener("keydown", handleEsc);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, "Save");
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, "Saving...");
+  const cardData = { name: cardNameInput.value, link: cardLinkInput.value };
+
+  api
+    .createCard(cardData)
+    .then((data) => {
+      const cardElement = getCardElement(data, { _id: "currentUserId" }); // Pass userInfo or mock it
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      evt.target.reset();
+    })
+    .catch((err) => {
+      console.error("Error creating card:", err);
+    })
+    .finally(() => {
+      setButtonText(submitBtn, "Save");
```

Comment:
`setButtonText(submitBtn, "Save");`

`"Save"` is not needed as the 2nd argument. There should be `false`


```js

    .finally(() => {
      setButtonText(submitBtn, false);
    });
```